### ISSUE TITLE
Right to left language support

### ISSUE THREAD

**Issue by meichthys** [TriliumNext/Notes#97]
zadam/trilium#623
zadam/trilium#1923

**Comment by biscqit** [comment]
this

**Comment by eliandoran** [comment]
@biscqit , are you willing to help us with translating the messages to an RTL language such as Arabic?

**Comment by biscqit** [comment]
how'd I do that?

**Comment by eliandoran** [comment]
@biscqit , depends on your level of experience.
Our translations are currently stored in a JSON format, should be relatively easy to modify provided you have experience with Git.
If not, we can try looking into adding support for a translation platform, provided there is a commitment from you to translate it.

**Comment by MohsenSameti** [comment]
ckeditor5 has support for rtl. can i enable that somehow in the editor? do not care much about the language of the frontend, just the editor

**Comment by eliandoran** [comment]
@MohsenSameti , what language would you need so that I can try adding support for it.

**Comment by MohsenSameti** [comment]
> [@MohsenSameti](https://github.com/MohsenSameti) , what language would you need so that I can try adding support for it.

persian.

although i mostly write in english, but sometomes i need to write in persian as well


@eliandoran

**Comment by eliandoran** [comment]
@MohsenSameti , can you have a look at the [demo editor for CKEditor](https://ckeditor.com/docs/ckeditor5/latest/features/language.html#demo)?

I need to understand whether the editing experience for RTL is OK for you. This plugin allows to switch languages, including Arabic.

What I find a bit strange is that the right-to-left text is not right-aligned (can be observed when deleting some characters):

![Image](https://github.com/user-attachments/assets/6e7ad09d-e74a-4d43-9f23-a04f0a108f95)

Contrast this with [setting the content language](https://ckeditor.com/docs/ckeditor5/latest/getting-started/setup/ui-language.html#demo-3), which applies the RTL globally to the document:

![Image](https://github.com/user-attachments/assets/e0ae28d5-3606-496c-85f1-da7ca726e821)

Let me know if the first demo is suitable.

**Comment by MohsenSameti** [comment]
> Is it more common to have a note that is a mix of RTL (Arabic, Persian, etc.) but also to contain LTR (English) text as opposed to having different notes (some RTL, some LTR)?


yes actually, i am a programmer and alot of words are used as their english not their translation, so most of my notes (which is related to programming is a mixture of both

**Comment by MohsenSameti** [comment]
> [@MohsenSameti](https://github.com/MohsenSameti) , can you have a look at the [demo editor for CKEditor](https://ckeditor.com/docs/ckeditor5/latest/features/language.html#demo)?

i couldn't understand what to do. the link you provided does not provide a demo, i tries to install it but it's not editable, and p tag i changed does not render text correctly.

![Image](https://github.com/user-attachments/assets/4af9aef5-cd1c-4bd0-a682-315b870212ce)

same thing in the live demo: https://ckeditor.com/ckeditor-5/demo/feature-rich/

@eliandoran

**Comment by eliandoran** [comment]
@MohsenSameti , not sure to understand your problem. The demo is [right here](https://ckeditor.com/docs/ckeditor5/latest/features/language.html#demo).

![Image](https://github.com/user-attachments/assets/7bb14368-d7fa-4ace-81d8-c900e3be5bf5)

Unfortunately it contains only Arabic but I guess it should be OK for testing just the RTL support.

You have a language switcher that is here:

![Image](https://github.com/user-attachments/assets/87883c89-e0dc-4643-bedd-21f690902775)

As I mentioned, I think there's a problem with the alignment of text for this approach. I guess we need to specify on a per-note basis whether we should align the text from right to left or from left to right.

**Comment by eliandoran** [comment]
Preliminary support for RTL added in https://github.com/TriliumNext/Notes/pull/1319.

**Comment by MohsenSameti** [comment]
> [@MohsenSameti](https://github.com/MohsenSameti) , not sure to understand your problem. The demo is [right here](https://ckeditor.com/docs/ckeditor5/latest/features/language.html#demo).


ah, i saw it now. previously the page did not load the editor probably.  the demo works, thanks. it is not right aligned, but sentences containing both english and persian are rendered correctly

it is okay if it only contains arabic. difference between that and persian is just 4 letters (which if your font support it it's okay

@eliandoran

**Comment by eliandoran** [comment]
@MohsenSameti , I've added basic LTR support in the latest beta which allows setting a note to a RTL language. This will inform the editor and properly align the text as well. The only downside is that it doesn't allow mixing RTL and LTR text since the demo provided previously did not properly align the text.

Let me know if it works out for you and feel free to open issues for it.
